lang/cs/Org.Apache.REEF.Network/Group/Driver/Impl/CommunicationGroupDriver.cs

```diff
@@ -119,12 +121,23 @@ public ICommunicationGroupDriver AddBroadcast<T>(string operatorName, string mas
                 topology = new FlatTopology<T>(operatorName, _groupName, spec.SenderId, _driverId,
                     spec);
             }
-            else
+            else if(topologyType == TopologyTypes.Tree)
             {
                 topology = new TreeTopology<T>(operatorName, _groupName, spec.SenderId, _driverId,
                     spec,
                     _fanOut);
             }
+            else if (topologyType == TopologyTypes.WritableFlat)
```

Comment:
Can you elaborate on this. Do you mean add functions like AddWritableBoradcast, AddWritableReduce?